src/crypto/NoteEncryption.hpp

```diff
@@ -0,0 +1,87 @@
+/*
+See the Zcash protocol specification for more information.
+https://github.com/zcash/zips/blob/master/protocol/protocol.pdf
+
+Zcash allows users to send and receive value using notes. In order
+to protect the privacy of the parties involved, this note is not
+public, and for verification purposes, is cryptographically
+committed. This commitment is public, and lives in the transaction.
+
+However, recipients must be able to obtain values of the note,
+including the value `v`, a secret `rho`, and the trapdoor `r`
+otherwise they will be unable to spend it. In addition, we would
+like to send a 64-byte memo to the recipient in the transaction.
+
+In order to do this privately, we require the use of an IND-CCA2
+secure encryption scheme. However, off-the-shelf schemes were
+poorly specified or had inefficient encodings in all of the known,
+credible implementations.
+
+In this scheme, we use an ephemeral Curve25519 keypair to
+construct a Diffie-Hellman secret with the recipient public key.
+This secret, the public ephemeral key, the public key of the
+recipient and a nonce (for indistinguishability of ciphertexts
+using the same ephemeral key) are used for a KDF. The key is then
```

Comment:
The parenthetical remark is confusing. I'd write it as:

This secret, the public ephemeral key, the public key of the recipient, a seed, and a nonce are fed into a KDF to derive the symmetric encryption key. The nonce is necessary because there are two encryptions within a JoinSplit, which have the same seed and ephemeral key pair, and may be to the same recipient public key.

Or something like that. Maybe just delete this comment so people are forced to refer to the spec?